File: week-7/day32/set1/Services/ContactService.cs
using DI.Models;
namespace DI.Services
{
    public class ContactService :IContactService
    {
        private static List<ContactInfo> contacts = new List<ContactInfo>();

        public List<ContactInfo> GetAllContacts()
        {
            return contacts;
        }
        public ContactInfo GetContactById(int id)
        {
            return contacts.FirstOrDefault(c => c.Id == id);

        }
        public void AddContacts(ContactInfo ci)
        {
            contacts.Add(ci);
        }

    }
}

File: week-11/day50/day50/src/app/Services/contact.service.ts
import { Injectable } from '@angular/core';
import { Contact } from '../Models/contacts';

@Injectable({
  providedIn: 'root'
})
export class ContactService {

  private contacts: Contact[] = [
    { id: 1, name: 'John Doe', email: 'john@gmail.com', phone: '9876543210' },
    { id: 2, name: 'Jane Smith', email: 'jane@gmail.com', phone: '9123456780' }
  ];

  constructor() {}

  getContacts(): Contact[] {
    return this.contacts;
  }

  addContact(contact: Contact): void {
    this.contacts.push(contact);
  }

  getContactById(id: number): Contact | undefined {
    return this.contacts.find(c => c.id === id);
  }
}